### ISSUE TITLE
Update Avy version in next release

### ISSUE THREAD

**Issue by quicknir** [syl20bnr/spacemacs#6206]
Not sure if this happens automatically or not, but avy was updated recently (couple of days ago), would be nice to see this update made available via spacemacs, cheers.

**Comment by d12frosted** [comment]
> Update Avy

What do you mean by this? If you wish to update any package - read [this](https://github.com/syl20bnr/spacemacs/blob/develop/doc/DOCUMENTATION.org#update-packages) section of documentation.

**Comment by quicknir** [comment]
If I update a package on my own though, it's not a) tested as part of the spacemacs distro, and b) when I go to another machine, how will that be persisted? The only thing synced between machines currently is my .spacemacs.

**Comment by d12frosted** [comment]
`~/.emacs.d/elpa` is folder containing all third-party packages that are downloaded during installation process. This directory is ignored by git. You can think of it as the artefacts. On new machine packages will be downloaded and byte-compiled again. If you wish to share them - you can do so, but spacemacs doesn't provide such functionality out of box, because it's needed only in **rare** cases (and I've never heard of such cases, but can imagine). 

> tested as part of the spacemacs distro

Spacemacs doesn't specify package versions to use. `package.el` downloads the latest available version. So when package is updated and brings some breaking changes - we just roll out new version of Spacemacs with fixes. One can use Stable MELPA as packages repository to avoid such situations, but Stable MELPA has a tradeoff - not all packages are there and some of packages are VERY outdated. But it's common in Emacs world (pretty much like in Arch Linux world) to depend on the very last version of package. If someone encounters a problem with some packages - the first thing to do is to update and check if it solves the problem.

TL;DR - just update as often as you wish, but if you encounter any problems with packages - try to update them. If it doesn't help - you could even try nuking `~/.emacs.d/elpa` dir and reinstall them again. Takes time but sometimes fixes some bugs (this happens pretty rarely).

I hope I answered your questions :)

**Comment by quicknir** [comment]
You did indeed, definitely quite a bit of misunderstanding on my part of spacemacs relationship to package versioning. Thanks for your patience and a great explanation!

**Comment by d12frosted** [comment]
You're welcome :)
